source/libs/cache.ts

```diff
@@ -1,73 +1,64 @@
+export async function has(key: string): Promise<boolean> {
+	const cachedKey = `cache:${key}`;
+	const values = await storage.get(cachedKey);
+	return values[cachedKey] !== undefined;
 }
 
-export async function get<TValue extends any = any>(key: string): Promise<TValue | undefined> {
-	const value = await browser.runtime.sendMessage({
-		key,
-		code: 'get-cache'
-	});
-
+export async function get<TValue extends unknown = unknown>(key: string): Promise<TValue | undefined> {
+	const cachedKey = `cache:${key}`;
+	const values = await storage.get(cachedKey);
+	const value = values[cachedKey];
 	// If it's not in the cache, it's best to return "undefined"
-	if (value === null || value === undefined) {
+	if (value === undefined) {
+		return undefined;
+	}
+
+	if (value.expiration > Date.now()) {
+		await storage.remove(cachedKey);
 		return undefined;
 	}
 
-	return value;
+	console.log('CACHE: found', key, value.data);
+	return value.data;
 }
 
-export function set<TValue extends any = any>(key: string, value: TValue, expiration?: number /* in days */): Promise<any> {
-	return browser.runtime.sendMessage({
-		key,
-		value,
-		expiration,
-		code: 'set-cache'
+export function set<TValue extends unknown = unknown>(key: string, value: TValue, expiration: number /* in days */): Promise<void> {
+	console.log('CACHE: setting', key, value);
+	const cachedKey = `cache:${key}`;
+	return storage.set({
+		[cachedKey]: {
+			data: value,
+			expiration: Date.now() + (1000 * 3600 * 24 * expiration)
+		}
 	});
 }
 
-/* Accept messages in background page */
-if (location.pathname === '/_generated_background_page.html') {
-	browser.runtime.onMessage.addListener(async (request: CacheRequest) => {
-		if (!request) {
-			return;
+async function purge(): Promise<void> {
+	const values = await storage.get();
+	const removableItems = [];
+	for (const [key, value] of Object.entries(values)) {
+		if (key.startsWith('cache:') && (value as CacheItem).expiration > Date.now()) {
+			removableItems.push(key);
 		}
+	}
 
-		const {code, key, value, expiration} = request;
-		if (code === 'get-cache') {
-			const [cached] = document.cookie.split('; ')
-				.filter(item => item.startsWith(key + '='));
-			if (cached) {
-				const [, value] = cached.split('=');
-				console.log('CACHE: found', key, value);
-				return JSON.parse(value);
-			}
-
-			console.log('CACHE: not found', key);
-		} else if (code === 'set-cache') {
-			console.log('CACHE: setting', key, value);
+	if (removableItems.length > 0) {
+		await storage.remove(removableItems);
+	}
+}
 
-			// Store as JSON to preserve data type
-			// otherwise Booleans and Numbers become strings
-			document.cookie = `${key}=${JSON.stringify(value)}; max-age=${expiration ? expiration * 3600 * 24 : ''}`;
-		}
-	});
+// Automatically clear cache every day
+if (isBackgroundPage()) {
+	// Call immediately to ensure the cache gets purged at some point.
+	setTimeout(purge, 60000);
+	setInterval(purge, 1000 * 3600 * 24);
 }
```

Comment:
I'd add a `export default {has, get, set}` instead of exporting each method separately, so it can be used as 

```js
import cache from './libs/cache'
```